File: 50 Coding Challenge Part I/TypeScript/Challenge15.ts
// Challenge 15: Create a function that will find the
// nth Fibonacci number using recursion.

function findFibonacci( num: number): number {
    if (num == 0){
        return 0;
    }
    if (num == 1){
        return 1;
    }

    return findFibonacci(num - 1) + findFibonacci(num - 2)
}

// Driver Method
console.log(findFibonacci(10))

File: 50 Coding Challenge Part I/dart/lib/challenge15.dart
/*  
  Coding Challenge 15: Create a function that will find the nth Fibonacci number using recursion.
  Reminder:
  F(0) = 0
  F(1) = 1
  F(n) = F(n-1) + F(n-2)

  @author: mofirojean
*/

int findFibonacci(int n) {
  if (n == 0) {
    return n;
  }

  if (n == 1) {
    return n;
  }

  return findFibonacci(n - 1) + findFibonacci(n - 2);
}
